src/NumericalAlgorithms/Interpolation/RegularGridInterpolant.hpp

```diff
@@ -0,0 +1,240 @@
+    const Variables<TagsList>& vars) const noexcept {
+  // For matrix multiplication of Interp . Source = Result:
+  //   matrix Interp is m rows by k columns
+  //   matrix Source is k rows by n columns
+  //   matrix Result is m rows by n columns
+  const size_t m = interpolation_matrices_[0].rows();
+  const size_t k = interpolation_matrices_[0].columns();
+  const size_t n = vars.number_of_independent_components;
+  ASSERT(k == vars.number_of_grid_points(),
+         "Number of grid points in source 'vars', "
+             << vars.number_of_grid_points()
+             << ",\n disagrees with the size of the source_mesh, " << k
+             << ", that was passed into the constructor");
+  if (result->number_of_grid_points() != m) {
+    *result = Variables<TagsList>(m, 0.0);
+  }
+  dgemm_('n', 'n', m, n, k, 1.0, interpolation_matrices_[0].data(), m,
+         vars.data(), k, 0.0, result->data(), m);
+}
+
+template <>
+template <typename TagsList>
+void RegularGrid<2>::interpolate(
+    const gsl::not_null<Variables<TagsList>*> result,
+    const Variables<TagsList>& vars) const noexcept {
+  // source_mesh extents: (sx, sy)
+  // target_mesh extents: (tx, ty)
+  // number of variables: n
+  const size_t tx = interpolation_matrices_[0].rows();
+  const size_t sx = interpolation_matrices_[0].columns();
+  const size_t ty = interpolation_matrices_[1].rows();
+  const size_t sy = interpolation_matrices_[1].columns();
+  const size_t n = vars.number_of_independent_components;
+  ASSERT(sx * sy == vars.number_of_grid_points(),
+         "Number of grid points in source 'vars', "
+             << vars.number_of_grid_points()
+             << ",\n disagrees with the size of the source_mesh, " << sx * sy
+             << ", that was passed into the constructor");
+  if (result->number_of_grid_points() != tx * ty) {
+    *result = Variables<TagsList>(tx * ty, 0.0);
+  }
+
+  // Note: The sequence of DGEMM and transpose calls needs several temporaries.
+  //       Here we loop over the variable's components (`var`), so that we
+  //       operate one DataVector at a time, with temporaries being DataVectors.
+  //       The algorithm _can_ be rewritten so each DGEMM call works on the
+  //       entire Vars at once, but then each temporary must become a Variables.
+  //       Furthermore, looping over variable components is still necessary in
+  //       the forward and backward transpose operations.
+  //       (Note: to move the DGEMM calls out of the loop, simply take the 4th
+  //        argument (number of rol/col multiplies) from, e.g., sy -> sy * n.)
+  const size_t source_size = sx * sy;
+  const size_t temp_size = tx * sy;
+  const size_t target_size = tx * ty;
+  // By making buffer1 large enough to hold two different stages in the
+  // computation, we can reuse it and avoid the need to allocate a 3rd buffer.
+  DataVector buffer1(std::max(temp_size, target_size));
+  DataVector buffer2(temp_size);
```

Comment:
Here I could optimize a little by joining these two allocations into one. Can do this either with the "temp" Variables approach (but then both buffers would have to be same size, leading to more allocation than strictly necessary), or pointers into malloc'd block. Thoughts?